### PR TITLE
Suggestions for community nodes repo

### PR DESCRIPTION
**Pull request by ivov**

### CODE REVIEW

**Review by BHesseldieck — APPROVED** [review]

**Review by krynble — CHANGES_REQUESTED** [review]

File: packages/cli/src/CommunityNodes/helpers.ts [hunk 1]
```diff
@@ -1,71 +1,89 @@
...
 	NPM_COMMAND_TOKENS,
 	NPM_PACKAGE_STATUS_GOOD,
 	RESPONSE_ERROR_MESSAGES,
+	UNKNOWN_FAILURE_REASON,
 } from '../constants';
-import { NpmPackageStatusCheck, NpmUpdatesAvailable, ParsedNpmPackageName } from '../Interfaces';
 import { InstalledPackages } from '../databases/entities/InstalledPackages';
 import config from '../../config';
 
+import type { Npm } from '../Interfaces';
```
[the remark this answers is not in the document]
Comment (ivov) [reply packages/cli/src/CommunityNodes/helpers.ts]: Agree, changed to `CommunityPackages`.

File: packages/cli/src/CommunityNodes/helpers.ts [hunk 2]
```diff
@@ -158,47 +168,38 @@ export function matchMissingPackages(
 	return hydratedPackageList;
 }
 
-export async function checkPackageStatus(packageName: string): Promise<NpmPackageStatusCheck> {
-	// You can change this URL for testing - the default testing url below
-	// is a postman mock service
-	const n8nBackendServiceUrl = 'https://api.n8n.io/api/package';
+export async function checkNpmPackageStatus(packageName: string): Promise<Npm.PackageStatusCheck> {
+	// You can change this URL for testing
+	// This default testing URL is a Postman mock service
```
[the remark this answers is not in the document]
Comment (ivov) [reply packages/cli/src/CommunityNodes/helpers.ts]: Removed.

File: packages/cli/src/CommunityNodes/packageModel.ts [hunk 3]
```diff
@@ -4,24 +4,24 @@ import { Db } from '..';
...
 ): Promise<InstalledPackages | undefined> {
-	const installedPackage = await Db.collections.InstalledPackages.findOne(packageName, {
-		relations: ['installedNodes'],
-	});
-	return installedPackage;
+	return Db.collections.InstalledPackages.findOne(packageName, { relations: ['installedNodes'] });
+}
+
+export async function isPackageInstalled(packageName: string): Promise<boolean> {
+	return findInstalledPackage(packageName) !== undefined;
```
[the remark this answers is not in the document]
Comment (ivov) [reply packages/cli/src/CommunityNodes/packageModel.ts]: Great catch! Fixed.

File: packages/cli/src/api/nodes.api.ts [hunk 4]
```diff
@@ -1,295 +1,311 @@
...
-});
+const { PACKAGE_NOT_INSTALLED, PACKAGE_NAME_NOT_PROVIDED } = RESPONSE_ERROR_MESSAGES;
+
+export const nodesController = express.Router();
 
 nodesController.use((req, res, next) => {
 	if (!isAuthenticatedRequest(req) || req.user.globalRole.name !== 'owner') {
-		res.status(403).json({ status: 'error', message: 'Unauthorized' });
-		return;
+		throw new ResponseHelper.ResponseError('Unauthorized', undefined, 403);
```
[the remark this answers is not in the document]
Comment (ivov) [reply packages/cli/src/api/nodes.api.ts]: Restored and documented.

**Review by krynble — APPROVED** [review]

**Comment by janober** [comment]
Thanks a lot. Got merged.

**Comment by janober** [comment]
Got released with `n8n@0.189.0`

**Pull request merged**